astropy/coordinates/sky_coordinate.py

```diff
@@ -1687,7 +1696,18 @@ def from_name(cls, name, frame='icrs'):
 
         from .name_resolve import get_icrs_coordinates
 
-        icrs_coord = get_icrs_coordinates(name)
+        # if requested, first try extract coordinates from name. do this first
+        # since it may be much faster than sesame query
+        if use_parser:
+            from . import jparser
+            if jparser.search(name):
+                icrs_coord = jparser.to_skycoord(name, frame)
```

Comment:
Can you clarify why you're putting the j-parser here instead of in `get_icrs_coordinates` itself?  It seems more natural to me to do the latter (in which case you don't have to modify this at all other than to add the keyword and docstring entry.